validator/src/scheduler/mod.rs

```diff
@@ -51,9 +51,13 @@ pub use self::execution_result_ffi::{BatchResult, TransactionResult};
 pub struct ExecutionResults {
     pub beginning_state_hash: Option<String>,
     pub ending_state_hash: Option<String>,
-    pub batch_results: Vec<(String, Vec<TxnExecutionResult>)>,
+    pub batch_results: BatchExecutionResult,
 }
 
+pub type TxnExecutionResults = Option<Vec<TxnExecutionResult>>;
+
+pub type BatchExecutionResult = Vec<(String, TxnExecutionResults)>;
+
```

Comment:
Hmm, these hide some important details that make client code more confusing, namely that `BatchExecutionResult` is a collection and that `TxnExecutionResults` can be None.

I think:
```
pub type BatchExecutionResult = (String, Option<Vec<TxnExecutionResult>);
```
gives a more intuitive interface, because then `batch_results` still looks like a collection when you read the code (which you expect because the field is plural) and

Reply:
Sounds good.